crates/viewer/re_ui/src/list_item/list_item.rs

```diff
@@ -140,6 +142,18 @@ impl ListItem {
         self
     }
 
+    /// Controls whether [`Self`] calls [`ListItemContent::ui`] when the item is not currently
+    /// visible.
+    ///
+    /// Skipping rendering can increase performances for long lists that are mostly out of view, but
+    /// this will prevent any side effects from [`ListItemContent::ui`] from occurring. For this
+    /// reason, this is an opt-in optimization.
```

Comment:
to clarify, could such a side-effect change the vertical size of a list array? If not, then wouldn't it be reasonable to expect this to be opt-in? I mean we're passing a closure after all for drawing a thing - if the thing isn't on screen, why would you expect your closure to be called; lotsa place don't do that

Reply:
No, the vertical size is fully determined by `ListItem::height`. Also, this change only affects the list item itself, not the fact that it might have children in a collapsing section below it. So there cannot be any issue with vertical layout.

And you are probably right that I'm being overly defensive with this opt-in stuff. I really don't have a strong reason to _not_ make it opt-out (or even always on), except of being scared to break things in untested areas 😅

Reply:
hmmm I'd really like to make it opt-out, but we can also do this maybe just after the next release then if we're too scared of adding regressions

Reply:
Reminder: https://github.com/rerun-io/rerun/issues/8815